### PR TITLE
Add self registration flow in system tests

### PR DESCRIPTION
**Pull request by ymarcus93**
## Description of the change

Adds new codepath in system test to test for self-registration flow of agent

## Relevant release note information

Release Notes: Adds new codepath in system test to test for self-registration flow of agent

## Related JIRA tickets

Relates to JIRA: CWC-686

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [x] No

If yes, please explain:

### CODE REVIEW

File: src/system-tests/tests/system-test.types.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { DigitalOceanDistroImage } from '../digital-ocean/digital-ocean-ssm-target.service.types';
+
+/**
+ * SSMTestTargetAutoDiscovery represents an SSM test target that should be
+ * registered using the traditional, all-in-bash autodiscovery script that is
+ * retrieved from the backend.
+ */
+export type SSMTestTargetAutoDiscovery = {
+    type: 'old';
```
Comment (lipmas) [inline src/system-tests/tests/system-test.types.ts:9]: installType: 'autodiscovery'

File: src/system-tests/tests/system-test.types.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    dropletImage: DigitalOceanDistroImage;
+}
+
+/**
+ * SSMTestTargetSelfRegistrationAutoDiscovery represents an SSM test target that
+ * should be registered using the new, self-registration flow built into the
+ * agent itself.
+ */
+export type SSMTestTargetSelfRegistrationAutoDiscovery = {
+    type: 'new';
```
Comment (lipmas) [inline src/system-tests/tests/system-test.types.ts:19]: installType: 'package-manager'

File: src/system-tests/tests/system-test.ts [hunk 3]
```diff
@@ -269,14 +331,19 @@ async function cleanupDOTestTargets() {
...
 }
 
-async function setupSystemTestApiKey() {
-    const apiKeyName = `system-test-${systemTestUniqueId}-api-key`;
-    systemTestApiKey = await apiKeyService.createNewApiKey({ name: apiKeyName, isRegistrationKey: false });
-    console.log('created api key ' + systemTestApiKey.apiKeyDetails.id);
+async function setupSystemTestApiKeys() {
+    const restApiKeyName = `system-test-${systemTestUniqueId}-api-key`;
+    systemTestRESTApiKey = await apiKeyService.createNewApiKey({ name: restApiKeyName, isRegistrationKey: false });
+    console.log('Created REST api key ' + systemTestRESTApiKey.apiKeyDetails.id);
```
Comment (lipmas) [inline src/system-tests/tests/system-test.ts:337]: why dont we use logger for these as well

File: src/system-tests/tests/system-test.ts [hunk 4]
```diff
@@ -201,24 +210,77 @@ async function createDOTestClusters() {
     await checkAllSettledPromise(allClusterCreationResults);
 }
 
+function getSelfRegistrationAutodiscoveryScript(testTarget: SSMTestTargetSelfRegistrationAutoDiscovery, envId: string, registrationApiKeySecret: string): string {
```
Comment (lipmas) [inline src/system-tests/tests/system-test.ts:213]: `getPackageManagerRegistrationScript`

File: src/system-tests/tests/system-test.types.ts (hunk 2 above)
Comment (ymarcus93) [reply src/system-tests/tests/system-test.types.ts]: d17d23ef7f088c415c9af58aab3ed5a008b03f6f

File: src/system-tests/tests/system-test.ts (hunk 3 above)
Comment (ymarcus93) [reply src/system-tests/tests/system-test.ts]: c4a3de58b5d99abd6ca45d832b06eb33a9296905

File: src/system-tests/tests/system-test.ts (hunk 4 above)
Comment (ymarcus93) [reply src/system-tests/tests/system-test.ts]: d17d23ef7f088c415c9af58aab3ed5a008b03f6f

File: src/system-tests/tests/system-test.types.ts (hunk 1 above)
Comment (ymarcus93) [reply src/system-tests/tests/system-test.types.ts]: d17d23ef7f088c415c9af58aab3ed5a008b03f6f

**Comment by sidpremkumar** [comment]
Something is up with system tests: Even though the auto discovery targets come online: 
2021-12-21 at 7 48 AM

System tests will say all ssm targets failed: 
2021-12-21 at 7 48 AM

**Comment by sidpremkumar** [comment]
Also can u update the `Release Notes` field in the PR description 🙏

**Review by lipmas — APPROVED** [review]

**Comment by lipmas** [comment]
Rebased on master/API V2 changes

**Pull request merged**